Ficek_QuantiApp/Ficek_QuantiApp/Domain/Tests/RocketClientTests/RocketClient+happyTests.swift

```diff
@@ -9,72 +9,105 @@ import RequestBuilder
+      url: URL(string: "https://api.spacexdata.com/v4/rockets")!,
+      statusCode: 200,
+      httpVersion: nil,
+      headerFields: [
+        "\(HTTPHeaderName.acceptType.rawValue)": "\(AcceptTypeValue.json.rawValue)",
+        "\(HTTPHeaderName.contentType.rawValue)": "\(ContentTypeValue.json.rawValue)"
+      ]
+    )
     
-    var testScheduler: TestScheduler<
-        DispatchQueue.SchedulerTimeType,
-        DispatchQueue.SchedulerOptions
-    >! = DispatchQueue.test
+    let mockUrlRequester = URLRequester { _ in
+      Just((successResponse, mockResponse!))
+        .setFailureType(to: URLError.self)
+        .eraseToAnyPublisher()
+    }
     
-    override func invokeTest() {
-        withDependencies {
-            $0.heightConverter = .live()
-            $0.enginesConverter = .live()
-            $0.secondStageConverter = .live()
-            $0.firstStageConverter = .live()
-            $0.massConverter = .live()
-            $0.diameterConverter = .live()
-            $0.rocketConverter = .live()
-            $0.rocketsConverter = .live()
-        } operation: {
-          super.invokeTest()
-        }
-      }
+    let networkClient = NetworkClient(
+      urlRequester: mockUrlRequester,
+      networkMonitorClient: .live(onQueue: .main)
+    )
     
-    func test_fetchAllRockets_sucessful() throws {
-        let successResponse = try JSONEncoder().encode([RocketDTO].mock)
-        let mockResponse = HTTPURLResponse(
-              url: URL(string: "https://api.spacexdata.com/v4/rockets")!,
-              statusCode: 200,
-              httpVersion: nil,
-              headerFields: [
-                "\(HTTPHeaderName.acceptType.rawValue)": "\(AcceptTypeValue.json.rawValue)",
-                "\(HTTPHeaderName.contentType.rawValue)": "\(ContentTypeValue.json.rawValue)"
-              ]
-        )
-        
-        let mockUrlRequester = URLRequester { _ in
-              Just((successResponse, mockResponse!))
-                .setFailureType(to: URLError.self)
-                .eraseToAnyPublisher()
-        }
-        
-        let sut = withDependencies { dependency in
-            let networkClient = NetworkClient(
-                urlRequester: mockUrlRequester,
-                networkMonitorClient: .live(onQueue: .main)
-            )
-            
-            dependency.networkClient = networkClient
-        } operation: {
-            RocketClient.liveValue
+    let sut = withDependencies { dependency in
+      dependency.networkClient = networkClient
+    } operation: {
+      RocketClient.liveValue
+    }
+    
+    sut.fetchAllRocketsCombine()
+      .sink(
+        receiveCompletion: { completion in
+          switch completion {
+          case .finished: break
+          case let .failure(error):
+            XCTFail("\(error.causeName) - Fail")
+          }
+          expectation.fulfill()
+          
+        }, receiveValue: { rocket in
+          XCTAssertNoDifference(rocket, [Rocket].mock)
```

Comment:
Tohle neni uplne spravne. Pokud ti prijde prijde jen finished a zadna hodnota, tak ti ten test taky projde a to je false positive. Musis na `receiveValue` assertovat hodnotu (jak to delas) a pak jeste flipnout nejakej boolean, treba `valueReceived` na `true`. Pak po tom `waitForExpectations` jeste assertovat ten boolean. Zaroven se mi nelibi moc jak neprehledne resis to `failure` a `finished`. Pekne si izolovane ohandluj oba cases, na `failure` dej `XCTFail` a na `finished` dej `fulfill`. Nepouzivej to propadavani pres `break` a `continue`.

Reply:
Oprav to pls ve vsech testech.